crates/router/src/core/payments.rs

```diff
@@ -3569,6 +3929,192 @@ where
+            )
+        }
+    }?
+    .parse_value::<api::routing::StraightThroughAlgorithm>("RoutingAlgorithm")
+    .change_context(errors::ApiErrorResponse::InternalServerError)
+    .attach_printable("Failed to parse the straight through routing algorithm")?;
+
+    let (routable_connector_choice, should_perform_eligibility_check) =
+        routing::perform_routing_for_single_straight_through_algorithm(
+            &straight_through_routing_algorithm,
+            payment_data.get_creds_identifier(),
+        )
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Failed execution of straight through routing")?;
+
+    // Filter the network transaction id supported connectors from the routable connector choices.
+    let mit_connector = filter_eligible_connectors(
+        state,
+        payment_data,
+        key_store,
+        routable_connector_choice,
+        should_perform_eligibility_check,
+        eligible_routable_connectors.as_ref(),
+        business_profile,
+    )
+    .await?;
+
+    let mit_first_connector = mit_connector
+        .first()
+        .ok_or(errors::ApiErrorResponse::IncorrectPaymentMethodConfiguration)
+        .attach_printable(
+            "No eligible connector found for the network transaction id based mit flow",
+        )?;
+
+    // Set `NetworkMandateId` as the MandateId
+    payment_data.set_mandate_id(payments_api::MandateIds {
+        mandate_id: None,
+        mandate_reference_id: Some(mandate_reference_id),
+    });
+
+    // Set the card details received in the recurring details within the payment method data.
+    payment_data.set_payment_method_data(Some(
+        hyperswitch_domain_models::payment_method_data::PaymentMethodData::CardDetailsForNetworkTransactionId(card_details),
+    ));
+
+    // Construct routing data using the selected connector information, which will be set in the payment attempt.
+    let routing_data = storage::RoutingData {
+        routed_through: Some(mit_first_connector.connector_name.clone().to_string()),
+        merchant_connector_id: mit_first_connector.merchant_connector_id.clone(),
+        routing_info: storage::PaymentRoutingInfo {
+            algorithm: None,
+            pre_routing_results: None,
+        },
+        algorithm: Some(straight_through_routing_algorithm),
+    };
+    let encoded_info = routing_data
+        .routing_info
+        .encode_to_value()
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("error serializing payment routing info to serde value")?;
+
+    payment_data.set_connector_in_payment_attempt(routing_data.routed_through);
+
+    payment_data.set_merchant_connector_id_in_attempt(routing_data.merchant_connector_id);
+    payment_data.set_straight_through_algorithm_in_payment_attempt(encoded_info);
+    Ok(mit_first_connector.clone())
+}
+
+pub async fn filter_eligible_connectors<F, D>(
```

Comment:
function name can be better, this is more generic and what underlying functionality is not expressed fully